### PR TITLE
Assignment Three Finished

### PR DESCRIPTION
**Pull request by LBerkholtz**
Ready to be graded (I hope).

### CODE REVIEW

File: src/main/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplication.java [hunk 1]
```diff
@@ -43,7 +60,7 @@ public static void main(String[] args) throws ParseException {
                 startDate.setTime(startdateformat.parse(fromDate));
                 endDate.setTime(enddateformat.parse(untilDate));
 
-            } catch (ParseException e) {
+                } catch (ParseException e) {
 
                 e.printStackTrace();
```
Comment (smarks) [inline src/main/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplication.java:65]: yikes, there is that unfortunate error handling (or actually the lack of it). 
Let me know if you need help here understanding how to handle exceptions correctly

File: src/main/java/com/lberkholtz/advancedava/week3/service/BasicStockService.java [hunk 2]
```diff
@@ -14,10 +18,10 @@ public BasicStockService() {
...
- * Create a new <CODE>BasicStockService</CODE> instance
- *
- * @param stockService used to get actual stock data (external dependency)
- */
+    * Create a new BasicStockService instance
+    *
+    * @param stockService used to get the stockservice
+    */
 
 public BasicStockService(StockService stockService){
```
Comment (smarks) [inline src/main/java/com/lberkholtz/advancedava/week3/service/BasicStockService.java:26]: I don't understand this. why are you passing in a StockService to the BasicStockService constructor?

File: src/main/java/com/lberkholtz/advancedava/week3/service/StockServiceFactory.java [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 /**
  * Returns a concrete implementation of the
- * interface depending on
+ * interface depending on the parameter passed
  */
 public class StockServiceFactory{
     /**
      * Just coding for BasicStockService now, can add more types of StockService in the future.
-     * @param stockServiceType
+     * @param stockServiceType what type of stock service is needed. For now we just have Basic
```
Comment (smarks) [inline src/main/java/com/lberkholtz/advancedava/week3/service/StockServiceFactory.java:14]: recall from the chat, we don't want the client code (code calling our code) to know about the concrete types, just the interfaces. If this is not clear to you, set up a 1:1 with me and we can review.

File: src/test/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Tests the StockQuoteApplication class
+ */
+
+public class StockQuoteApplicationTest {
+    String[] args = {"APPL", "2018-01-01", "2018-01-10"};
+
+    @Test
+    public void main()throws ParseException {
+        StockQuoteApplication.main(args);
```
Comment (smarks) [inline src/test/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplicationTest.java:20]: use a better method name for the test. 
also, test for excepted exceptions like this:
  @Test(expected = IllegalArgumentException.class)
    public void testNegative() {
        App app = new App(null);
        assertNull("Verify the message is set correctly", app.getMessage());

**Review by smarks — APPROVED** [review]

File: src/main/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplication.java (hunk 1 above)
Comment (LBerkholtz) [reply src/main/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplication.java:65]: I ended up changing it to this so the program terminates with an error message. If there was input from the console I could ask for a different input, but we don't have that in this application, so I figured terminating with an error was best.  Let me know if this is OK.

                } catch (ParseException e) {

                System.out.println("Invalid dates. Program is terminating");
                System.exit(-1);

            }

File: src/main/java/com/lberkholtz/advancedava/week3/service/BasicStockService.java (hunk 2 above)
Comment (LBerkholtz) [reply src/main/java/com/lberkholtz/advancedava/week3/service/BasicStockService.java:26]: Good question! Makes no sense at all. I fixed it.

File: src/main/java/com/lberkholtz/advancedava/week3/service/StockServiceFactory.java (hunk 3 above)
Comment (LBerkholtz) [reply src/main/java/com/lberkholtz/advancedava/week3/service/StockServiceFactory.java:14]: I am going to have to set up a 1 on 1 to make sure I understand this. I think I'm missing something.

File: src/test/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplicationTest.java (hunk 4 above)
Comment (LBerkholtz) [reply src/test/java/com/lberkholtz/advancedava/week3/Main/StockQuoteApplicationTest.java:20]: Not sure about this. I'll have to talk to you about it in the 1 on 1.

**Pull request merged**